src/phase_1_image_preprocessing/scratch_directory_manager.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Nov 28 11:55:15 2018
+
+Scratch_Directory_Manager.py -add fname -remove fname -lookup fname --remove_all --verbose
+
+@author: mmacferrin
+"""
+
+#######################################################################################
+## Code for importing the parent directory in order to get the File_Locations object
+import os,sys,inspect
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+parentdir = os.path.dirname(currentdir)
+sys.path.insert(0,parentdir)
+## Also add the utility folder for the csv names
+utilitydir = os.path.join(parentdir, "utilities")
+sys.path.insert(0,utilitydir)
+#######################################################################################
+
+from file_locations import ASIFT_Scratch_Dir, ASIFT_Scratch_Logfile
+from csv_auto_reader import read_simple_csv
+from csv_writer import write_csv
+
+import hashlib
+import os
+import shutil
+import numpy
+import argparse
+
+class SCRATCH_Manager:
+
+    def __init__(self, scratch_dir=ASIFT_Scratch_Dir, scratch_logfile=ASIFT_Scratch_Logfile):
+        self.scratch_dir = scratch_dir
+        self.logfile = scratch_logfile
+        self.logfile_data = None
+        self.logfile_numpy_dt = None
+        return
+
+    def hash_filename(self, fname, remove_path=True):
```

Comment:
Not a security issue, no. Just getting around having absurdly-super-long directory names by using a hashed string. Since some of the files (WorldView especially) can have very long names that only differ at the end of the string but have the same (say) initial 100 characters, I opted to use that rather than just the first ~40-ish characters which could cause conflicts. If a different method is preferred to generate unique directory names, I'm not married to that one, it's just one that worked.

Reply:
I see! Thank you!